etna/transforms/decomposition/base_change_points.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import Tuple
+from typing import Type
+
+import pandas as pd
+from ruptures.base import BaseEstimator
+from ruptures.costs import CostLinear
+from sklearn.base import RegressorMixin
+
+TTimestampInterval = Tuple[pd.Timestamp, pd.Timestamp]
+TDetrendModel = Type[RegressorMixin]
+
+
+class BaseChangePointsModelAdapter(ABC):
+    """BaseChangePointsModelAdapter is the base class for transforms with change points."""
+
+    def __init__(self, change_point_model: BaseEstimator, **change_point_model_predict_params):
+        """Init BaseChangePointsModelAdapter.
+
+        Parameters
+        ----------
+        change_point_model:
+            model to get change points
+        change_point_model_predict_params:
+            params for ``change_point_model.predict`` method
+        """
+        self.change_point_model = change_point_model
+        self.change_point_model_predict_params = change_point_model_predict_params
+
+    @abstractmethod
+    def get_change_points(self, df: pd.DataFrame, in_column: str) -> List[pd.Timestamp]:
+        """Find change points in given dataframe and column."""
+        pass
+
+    @staticmethod
+    def _build_intervals(change_points: List[pd.Timestamp]) -> List[TTimestampInterval]:
+        """Create list of stable intervals from list of change points."""
+        change_points = sorted(change_points)
+        left_border = pd.Timestamp.min
+        intervals = []
+        for point in change_points:
+            right_border = point
+            intervals.append((left_border, right_border))
+            left_border = right_border
+        intervals.append((left_border, pd.Timestamp.max))
+        return intervals
+
+
+class RupturesChangePointsModel(BaseChangePointsModelAdapter):
+    """RupturesChangePointsModel is the base class for holding ruptures models."""
+
+    def __init__(self, change_point_model: BaseEstimator, **change_point_model_predict_params):
+        """Init RupturesChangePointsModel.
+
+        Parameters
+        ----------
+        change_point_model:
+            model to get change points
+        change_point_model_predict_params:
+            params for ``change_point_model.predict`` method
+        """
+        super(RupturesChangePointsModel, self).__init__(
+            change_point_model=change_point_model,
+            **change_point_model_predict_params,
+        )
+
+    @staticmethod
+    def find_change_points_segment(
```

Comment:
I am not sure that we actually need this method. It is an implementation of `get_change_points`